wallet.sh

```diff
@@ -0,0 +1,189 @@
+#!/bin/bash
+
+# set default config
+# this makes running bitcoin-cli interactively much easier
+
+echo "
+chain=${CHAIN}
+rpcuser=${RPC_USER}
+rpcpassword=${RPC_PASSWORD}
+rpcallowip=0.0.0.0/0
+
+[${CHAIN}]
+rpcbind=0.0.0.0
+">~/.bitcoin/bitcoin.conf
+
+ensure_bitcoin_is_running() {
+  process_check=`ps aux | grep -v grep | grep bitcoind`
+  if [[ -z "${process_check}" ]]; then
+    echo "Bitcoind seems to have crashed, we are going to ensure no lock exists on the process and restart the daemon."
+    remove_lock_if_exists
+    start_bitcoind_daemon
+  fi
+}
+
+check_bitcoin_is_running() {
+  process_check=`ps aux | grep -v grep | grep bitcoind`
+  echo "${process_check}"
+}
+
+remove_lock_if_exists() {
+  if [ -f "~/.bitcoin/bitcoind.pid" ]; then
+    rm -rf ~/.bitcoin/bitcoind.pid || echo "Failed to delete PID"
+  else
+    echo "PID Doesn't Exist"
+  fi
+
+  if [ -f "~/.bitcoin/${CHAIN}/.lock" ]; then
+   rm -rf ~/.bitcoin/${CHAIN}/.lock || echo "Failed to delete data lock"
+  else
+    echo "Failed to delete data lock"
+  fi
+}
+
+start_bitcoind_daemon() {
+  start_bitcoind -daemon
+}
+
+start_bitcoind() {
+  bitcoind \
+    -pid=${HOME}/.bitcoin/bitcoind.pid \
+    -listen=1 \
+    -server=1 \
+    -txindex=1 \
+    -deprecatedrpc=create_bdb \
+    -deprecatedrpc=warnings \
+    $@
+}
+
+stop_bitcoind_daemon() {
+  bitcoin_pid=$(pgrep bitcoind)
+  echo "Kill bitcoind with kill -SIGTERM"
+  kill -SIGTERM "$bitcoin_pid"
+  echo "bitcoind PID: ${bitcoin_pid}"
+  while kill -0 "$bitcoin_pid" 2> /dev/null; do
+    echo "Waiting for bitcoind process to stop."
+    check_bitcoin_is_running
+    sleep 1
+  done
+}
+
+wait_for_daemon_active() {
+  while true; do
+    check_bitcoin_is_running
+    if bitcoin-cli getblockchaininfo ; then
+      return
+    fi
+    echo "Waiting for bitcoind to start..."
+    sleep 5
+  done
+}
+
+get_current_height() {
+  if [[ -z $NETWORK_HEIGHT_URL ]]; then
+    case $CHAIN in
+      "main")
+      NETWORK_HEIGHT_URL=https://mempool.space/api/blocks/tip/height
+        ;;
+      "testnet3")
+        NETWORK_HEIGHT_URL=https://mempool.space/testnet/api/blocks/tip/height
+        ;;
+      "testnet4")
+        NETWORK_HEIGHT_URL=https://mempool.space/testnet4/api/blocks/tip/height
+        ;;
+      "signet")
+        NETWORK_HEIGHT_URL=https://mempool.space/signet/api/blocks/tip/height
+        ;;
+      "regtest")
+        echo 0
+        return
+        ;;
+      *)
+        echo "Unsupported chain: $CHAIN" >&2
+        return 1
+        ;;
+    esac
+  fi
+  curl -s ${NETWORK_HEIGHT_URL}
```

Comment:
A retry here may be also useful

Reply:
This will be called repeatedly in the `wait_for_network_sync` loop so if it fails once it shouldn't be that big of an issue imo